packages/@ama-sdk/core/src/plugins/session-id/session-id.request.ts

```diff
@@ -32,30 +33,41 @@ export class SessionIdRequest implements RequestPlugin {
   /**
    * The session ID used to track API calls
    */
-  public sessionId: string;
+  public sessionId?: string;
 
   /**
    * Constructor.
    * @param sessionIdHeader The request header in which the ID will be added.
    * @param activateRequestId Indicates if the request ID part should be added to the ID.
    */
   constructor(sessionIdHeader = 'Ama-Client-Ref', activateRequestId = true) {
+    // Declaration done first since generateSessionId uses the logger
     this.sessionIdHeader = sessionIdHeader;
     this.requestIdActivated = activateRequestId;
-    this.sessionId = this.generateSessionId();
   }
 
-  private logSessionId(sessionId: string, date: string) {
-    // eslint-disable-next-line no-console, no-restricted-syntax
-    console.info(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
-    // eslint-disable-next-line no-console, no-restricted-syntax
-    console.info(`Generated at: ${date}`);
+  private logSessionId(sessionId: string, date: string, logger?: Logger) {
+    if (logger?.info) {
+      logger.info(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
+      logger.info(`Generated at: ${date}`);
+    } else if (logger?.log) {
+      logger.log(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
+      logger.log(`Generated at: ${date}`);
+    } else {
+      console.info(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
+      console.info(`Generated at: ${date}`);
```

Comment:
```suggestion
    (logger?.info || logger?.log || console.info)(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
    (logger?.info || logger?.log || console.info)(`Generated at: ${date}`);
```

Reply:
not sure to understand why it would not work :D.
Can you give another try, we never now :D

Reply:
I tried again, and it seems only to work when I use the `bind` function:
```
(logger?.info || logger?.log || console.info).bind(logger || console)(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
(logger?.info || logger?.log || console.info).bind(logger || console)(`Generated at: ${date}`);
```
is this better do you think?